elastalert/elastalert.py

```diff
@@ -686,6 +687,9 @@ def get_segment_size(self, rule):
 
     def get_query_key_value(self, rule, match):
         # get the value for the match's query_key (or none) to form the key used for the silence_cache.
+        # Flatline ruletype sets "key" instead of the actual query_key
+        if isinstance(rule['type'], FlatlineRule) and 'key' in match:
+            return match['key']
```

Comment:
I think you should still go through `get_named_key_value`, but pass `key` instead of `query_key`, so that you can get the benefits that come with it (unicode stuff, using `lookup_es_key` for better handling of complex keys)